tripal_chado/includes/tripal_chado.field_storage.inc

```diff
@@ -727,6 +727,16 @@ function tripal_chado_field_storage_query($query) {
       } // end foreach ($field['bundles']['TripalEntity'] as $bundle_name) {
     } // end if ($sort['type'] == 'field') {
   } // end foreach ($query->order as $index => $sort) {
+  
+  
+  // Only include records that are deleted.  Tripal doesn't keep track of
+  // records that are deleted that need purging separately so we can do nothing
+  // with this.
+  if (isset($query->deleted)) {
```

Comment:
can you elaborate on the point of this?  Will we some day be able to something with this?

Reply:
Drupal fields create their own tables for each field.  And each table has a 'deleted' column.  I believe this allows Drupal to mark field values attached to entities as deleted without taking the time to remove them.  Then during cron runs it will purge those deleted items.  That's what's actually happening here, Drupal tries to purge the first 10 deleted values when you delete the field from the content type.  However, our Tripal fields do not keep a deleted state.  We don't want the field query to return anything if the query is to find fields that are marked as deleted because our TripalFields don't do that.  So, it should always return nothing.